### PR TITLE
Implement databricks connector as a copy destination

### LINKED ISSUE #2075: New destination: Databricks Delta Lake
## Tell us about the new integration you’d like to have
I would like to see Delta Lake support through Apache or Databricks Spark

## Describe the context around this new integration
Outputting CDC , Kafka topics ect.. to a data lake. 

## Describe the alternative you are considering or using
At the moment Fivetran supports this feature so I will use both products as needed.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368148044757) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by tuliren**
- Change the implementation from `JdbcDestination` to `CopyDestination`.
- Follow up PR for #5629.

### CODE REVIEW

File: airbyte-integrations/connectors/destination-databricks/build.gradle [hunk 1]
```diff
@@ -15,6 +15,14 @@ dependencies {
     implementation project(':airbyte-integrations:bases:base-java')
     implementation files(project(':airbyte-integrations:bases:base-java').airbyteDocker.outputs)
     implementation project(':airbyte-integrations:connectors:destination-jdbc')
+    implementation project(':airbyte-integrations:connectors:destination-s3')
```
Comment (tuliren) [inline airbyte-integrations/connectors/destination-databricks/build.gradle:18]: Let's depend on the s3 destination for now and DRY it later.

File: airbyte-integrations/connectors/destination-databricks/src/main/java/io/airbyte/integrations/destination/databricks/DatabricksDestination.java [hunk 2]
```diff
@@ -25,93 +25,60 @@
...
-  public static void getDriver() throws MalformedURLException, ClassNotFoundException {
-    File driverJar = new File("/Users/phlair/Downloads/SparkDriver/SparkJDBC42.jar");
-    URL jarUrl = new URL("jar", "", "file:" + driverJar.getAbsolutePath() + "!/");
-    URLClassLoader myLoader = new URLClassLoader(new URL[] { jarUrl } );
-    myLoader.loadClass(DRIVER_CLASS);
-  }
+  private static final String DRIVER_CLASS = "com.simba.spark.jdbc.Driver";
 
   @Override
-  public AirbyteConnectionStatus check(JsonNode config) {
```
Comment (tuliren) [inline airbyte-integrations/connectors/destination-databricks/src/main/java/io/airbyte/integrations/destination/databricks/DatabricksDestination.java:59]: The `CopyDestination` can check or create the SQL tables by calling the abstract method. So we only need to check the persistence in this implementation.

File: airbyte-integrations/connectors/destination-databricks/src/main/java/io/airbyte/integrations/destination/databricks/DatabricksStreamCopier.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+import java.sql.Timestamp;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This implementation is similar to {@link io.airbyte.integrations.destination.jdbc.copy.s3.S3StreamCopier}.
+ * The difference is that this implementation creates Parquet staging files, instead of CSV ones.
+ */
+public class DatabricksStreamCopier implements StreamCopier {
```
Comment (tuliren) [inline airbyte-integrations/connectors/destination-databricks/src/main/java/io/airbyte/integrations/destination/databricks/DatabricksStreamCopier.java:26]: We can focus on implementing the methods in this class to transfer the data from S3 to Databricks.

File: airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/copy/StreamCopier.java [hunk 4]
```diff
@@ -36,7 +37,7 @@
   /**
    * Writes a value to a staging file for the stream.
    */
-  void write(UUID id, String jsonDataString, Timestamp emittedAt) throws Exception;
+  void write(UUID id, AirbyteRecordMessage recordMessage) throws Exception;
```
Comment (tuliren) [inline airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/copy/StreamCopier.java:40]: This interface change is necessary to drop in the `S3ParquetWriter` directly for convenience.

File: airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/copy/StreamCopierFactory.java [hunk 5]
```diff
@@ -28,17 +28,25 @@
...
 import io.airbyte.protocol.models.DestinationSyncMode;
 
 public interface StreamCopierFactory<T> {
 
   StreamCopier create(String configuredSchema,
                       T config,
                       String stagingFolder,
-                      DestinationSyncMode syncMode,
-                      AirbyteStream stream,
+                      ConfiguredAirbyteStream configuredStream,
```
Comment (tuliren) [inline airbyte-integrations/connectors/destination-jdbc/src/main/java/io/airbyte/integrations/destination/jdbc/copy/StreamCopierFactory.java:39]: This interface change is necessary to reuse the `S3Writer`.

**Review by Phlair — APPROVED** [review]

**Comment by Phlair** [comment]
@tuliren I'll merge this in and then work off it

**Pull request merged**